discovery-provider/src/tasks/entity_manager/delegation.py

```diff
@@ -75,35 +101,77 @@ def validate_delegation_tx(
             not in params.existing_records[EntityType.APP_DELEGATE]
         ):
             raise Exception(
-                f"Invalid Delegation transaction, delegate address {metadata['delegate_address']} does not exist"
+                f"Invalid Create Delegation transaction, delegate address {metadata['delegate_address']} does not exist"
             )
         if (
             metadata["shared_address"].lower()
             in params.existing_records[EntityType.DELEGATION]
+            and not params.existing_records[EntityType.DELEGATION][
```

Comment:
Re: your question @sliptype - this line effectively allows us to reactivate delegations. I.e. now you can re-create an existing delegation if it is in the revoked state.